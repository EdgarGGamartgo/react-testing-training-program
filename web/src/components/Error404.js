import React from 'react'

const styles = {
    background: '#232323',
    minHeight: '90vh',
    padding: '20px',
    color: 'white',
    textAlign: 'center',
    position: 'relative'
}

const styleH = {
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translateX(-50%) translateY(-50%)',
    margin: '0',
    marginRight: '-50%'
}

const styleB = {
    position: 'absolute',
    top: '60%',
    left: '50%',
    transform: 'translateX(-50%) translateY(-60%)',
    margin: '0',
    marginRight: '-50%',
    color: '#F65261',
    border: '2px solid #F65261',
    backgroundColor: '#232323'
}

export const Error404 = () => {
    return (
        <div style={styles}>
            <h1 style={styleH}>PAGE NOT FOUND 404</h1>
            <button style={styleB}>GO BACK TO HOME</button>
        </div>
    ) 
}