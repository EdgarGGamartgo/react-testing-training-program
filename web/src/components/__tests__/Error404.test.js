import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { Error404 } from './../Error404'

afterEach(cleanup)

it('should take a snapshot of Error404 presentational component', () => {
    const { asFragment } = render(<Error404 />)

    expect(asFragment(<Error404 />)).toMatchSnapshot()
})
