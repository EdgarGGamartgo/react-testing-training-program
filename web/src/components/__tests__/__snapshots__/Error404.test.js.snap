// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot of Error404 presentational component 1`] = `
<DocumentFragment>
  <div
    style="background: rgb(35, 35, 35); min-height: 90vh; padding: 20px; color: white; text-align: center; position: relative;"
  >
    <h1
      style="position: absolute; top: 50%; left: 50%; transform: translateX(-50%) translateY(-50%); margin: 0px -50% 0px 0px;"
    >
      PAGE NOT FOUND 404
    </h1>
    <button
      style="position: absolute; top: 60%; left: 50%; transform: translateX(-50%) translateY(-60%); margin: 0px -50% 0px 0px; color: rgb(246, 82, 97); border: 2px solid #f65261; background-color: rgb(35, 35, 35);"
    >
      GO BACK TO HOME
    </button>
  </div>
</DocumentFragment>
`;
